taxonomy-editor-frontend/src/pages/editentry/ListTranslations.jsx

```diff
@@ -1,94 +1,351 @@
+    const [btnDisabled, setBtnDisabled] = useState(true) // For enabling or disabling Dialog button
+
+    // Helper functions for Dialog component
+    function handleClose() { setOpen(false); }
+    function handleOpen() { setOpen(true); }
+
+    // Used for addition of a translation language
+    function handleAddTranslation(key) {
+        const newRenderedTranslations = [...renderedTranslations, {'languageCode' : key, 'tags' : []}]
+        setRenderedTranslations(newRenderedTranslations);
+        key = 'tags_' + key; // LC must have a prefix "tags_"
         
-        // Get all tags and its corresponding language code
-        // Tagids need to be recomputed, so shouldn't be rendered
-        // Main language isn't considered, since it's rendered separately
- 
-        if (key.startsWith('tags') && 
-            !key.endsWith(nodeObject.main_language) && 
-            !key.includes('ids')) {
+        // Make changes to the parent NodeObject
+        setNodeObject(prevState => {
+            const newNodeObject = {...prevState};
+            newNodeObject[key] = [];
+            return newNodeObject
+        })
+        setOpen(false);
+    }
+
+    // Used for deleting a translation language
+    function handleDeleteTranslation(key) {
+        const newRenderedTranslations = renderedTranslations.filter(obj => !(key === obj.languageCode))
+        setRenderedTranslations(newRenderedTranslations);
+        key = 'tags_' + key; // LC must have a prefix "tags_"
+        
+        // Make changes to the parent NodeObject
+        setNodeObject(prevState => {
+            const newNodeObject = {...prevState};
+            delete newNodeObject[key];
+            return newNodeObject
+        })
+        setOpen(false);
+    }
+
+    // Changes the translations to be rendered
+    // Dependent on changes occuring in "originalNodeObject"
+    useEffect(() => {
+
+        // Main langauge tags are considered separately, since they have to be rendered first
+        let mainLangTags = []
+        let otherLangTags = []
+
+        Object.keys(originalNodeObject).forEach((key) => {
+        
+            // Get all tags and its corresponding language code
+            // Tagids need to be recomputed, so shouldn't be rendered
+            // Eg: tags_fr
+        
+            if (key.startsWith('tags') && !key.includes('ids')) {
+
+                // If tags are not for the main language, add them to otherLangTags
+                if (!key.endsWith(originalNodeObject.main_language)) {
 
-                // Slice the language code
-                let languageCode = key.slice(-2);
-                renderedTranslations[languageCode] = nodeObject[key]
+                    // Slice the language code
+                    let languageCode = key.slice(-2);
+                    let tobeInsertedObj = {'languageCode' : languageCode, 'tags' : []} // General format for storing tags for different lc's
+                    originalNodeObject[key].map((tag) => (
```

Comment:
nit: better to not mutate inside a `.map`, this could be `.forEach`. But better do no mutation at all, it can lead to bugs or unexpected behaviors. 
```
const newNode = {...originalNodeObject}
const tags = newNode[key].map(tag => { ... })
```
But... I don't see where the originalNodeObject is being set again ... is that part missing and we are rellying in the mutation?

Reply:
`originalNodeObject` is the result of the node fetched. It changes only when a new node is fetched from the backend API. 

The tags related to main_language and other languages are stored in two separate arrays `mainLangTags` and `otherLangTags`.

Reply:
anyways, if using `.map()` better to have the usual syntax `const something = arr.map( ...)`